### PR TITLE
Add tests and fix hasPauseAnnotation

### PR DESCRIPTION
**Pull request by tfussell**
## Checklist

- [x] Update changelog in CHANGELOG.md.
- [ ] Update roadmap in ROADMAP.md.

### CODE REVIEW

File: pkg/controller/controller.go [hunk 1]
```diff
@@ -599,12 +599,17 @@ func ProcessUpdate(ctx context.Context, obj interface{}, resources []resource.In
...
+func hasAnnotation(annotations map[string]string, targetKey string, targetValue string) bool {
+	if annotations == nil {
 		return false
 	}
 
-	return m[k] == v
+	actualValue, ok := annotations[targetKey]
+	if !ok {
+		return false
+	}
```
Comment (kopiczko) [inline pkg/controller/controller.go:610]: It seems to me that it fixes only the case when `hasAnnotation(m, "non-existent", "") is called. Was this the issue?

Comment (tfussell) [reply pkg/controller/controller.go:610]: Exactly. I added the unit tests before changing the code and confirmed that this was the only case that was failing.

**Comment by tfussell** [comment]
I'll be testing this today.

**Comment by fiunchinho** [comment]
What's missing to be able to merge this?

**Comment by tfussell** [comment]
> What's missing to be able to merge this?

I can't test it anywhere. AWS clusters were failing to start when I tried it on an aws-operator branch. KVM is not compatible with this major version of operatorkit (I'm trying to upgrade it to k8s 1.18 now). If you have a moment to test it in azure-operator it would be cool otherwise I'll do it in KVM when I'm able.

**Comment by fiunchinho** [comment]
I will use this PR for testing it https://github.com/giantswarm/azure-operator/pull/1103

**Comment by fiunchinho** [comment]
It's working fine in Azure. I would merge this.

**Review by fiunchinho — APPROVED** [review]
Nice!

**Pull request merged**